src/ff/prime_field.rs

```diff
@@ -112,6 +116,60 @@ macro_rules! field_impl {
                 *self == other.as_u128()
             }
         }
+
+        #[cfg(test)]
+        paste! {
+            #[cfg(test)]
+            mod [< $field:lower _tests >] {
+                use super::*;
+                use proptest::proptest;
+
+                #[test]
+                fn zero() {
+                    let prime = u128::from($field::PRIME);
+                    assert_eq!($field::ZERO, $field::from(prime), "from takes a modulus",);
+                    assert_eq!($field::ZERO, $field::ZERO + $field::ZERO);
+                    assert_eq!($field::ZERO, $field::ZERO - $field::ZERO);
+                    assert_eq!($field::from(prime - 1), $field::ZERO - $field::ONE);
+                    assert_eq!($field::ZERO, $field::ZERO * $field::ONE);
+                }
+
+                #[test]
+                fn ser_not_enough_capacity() {
+                    let mut buf = [0; 0];
+                    assert!(matches!(
+                        $field::ONE.serialize(&mut buf),
+                        Err(e) if e.kind() == std::io::ErrorKind::WriteZero));
```

Comment:
Want to do two tests with capacity 0 and capacity `size_of!(Self)` (or `Self::SIZE_IN_BYTES`, I guess) ?

Reply:
would write a prop test for that, too lazy creating two tests. btw it is pretty reliable as it fails with 100% probability if `buf_capacity` is allowed to be enough to fit field values